### PR TITLE
Adding Multiselect Functionality in Panel Location Selection

### PR DESCRIPTION
**Pull request by claalmve**
Added multi-select functionality in location selection. Visually demonstrates regions highlighted and allows for exports with multiple geometries.

Here is what the layout looks like:

**Panel view:**
Screenshot 2023-08-22 at 4 20 53 PM

**Loader view**
Screenshot 2023-08-22 at 4 23 58 PM

### CODE REVIEW

File: climakitae/core/data_load.py
Comment (claalmve) [inline climakitae/core/data_load.py:291]: I added `shape_indicies` as arguments into the `set_subarea` function (not formerly added), but I thought for clean variable referencing, it was good to pass the argument through with the function, instead of referencing the variable outside of the function itself. Let me know what you think, and/or if I should leave it as is instead.

File: climakitae/core/data_interface.py
Comment (claalmve) [inline climakitae/core/data_interface.py:267]: This is something I implemented to catch errors because of delays that were occurring when re-rendering the panel (sometimes `cached_area` would return None, but that would cause defining `shape_indicies` to fail). This error would occur when subsetting the data would be selected by none. I would love to get another pair of eyes to see if this is a good fix or if there is something more concrete I could implement instead.

**Review by vicford — CHANGES_REQUESTED** [review]
This is super cool to test @claalmve!! Tested in getting_started with no problems. 

However, looks like the warming levels panel is now breaking, as is AMY. Error below:
```File ~/climakitae/climakitae/core/data_load.py:685, in _get_data_one_var(selections)
    682     data_dict = {**data_dict, **data_dict2}
    684 # Merge individual Datasets into one DataArray object.
--> 685 da = _merge_all(selections=selections, data_dict=data_dict)
    687 # Set data attributes and name
    688 data_attrs = _get_data_attributes(selections)

File ~/climakitae/climakitae/core/data_load.py:569, in _merge_all(selections, data_dict)
    566 hist_keys = [one for one in data_dict.keys() if "historical" in one]
    567 if hist_keys:
    568     all_hist = xr.concat(
--> 569         [
    570             _process_dset(one, data_dict[one], selections)
    571             for one in data_dict.keys()
    572             if "historical" in one
    573         ],
    574         dim="member_id",
    575     )
    576 else:
    577     all_hist = None

File ~/climakitae/climakitae/core/data_load.py:570, in <listcomp>(.0)
    566 hist_keys = [one for one in data_dict.keys() if "historical" in one]
    567 if hist_keys:
    568     all_hist = xr.concat(
    569         [
--> 570             _process_dset(one, data_dict[one], selections)
    571             for one in data_dict.keys()
    572             if "historical" in one
    573         ],
    574         dim="member_id",
    575     )
    576 else:
    577     all_hist = None

File ~/climakitae/climakitae/core/data_load.py:438, in _process_dset(ds_name, dset, selections)
    435 dset = _time_slice(dset, selections)
    437 # Perform area subsetting
--> 438 dset = _spatial_subset(dset, selections)
    440 # Perform area averaging
    441 if selections.area_average == "Yes":

File ~/climakitae/climakitae/core/data_load.py:382, in _spatial_subset(dset, selections)
    367 def _spatial_subset(dset, selections):
    368     """Subset over spatial area
    369 
    370     Parameters
   (...)
    380         subsetted area of dset
    381     """
--> 382     ds_region = area_subset_geometry(selections)
    384     if ds_region is not None:  # Perform subsetting
    385         if selections.downscaling_method == ["Dynamical"]:

File ~/climakitae/climakitae/core/data_load.py:289, in area_subset_geometry(selections)
    286     ds_region = [geom]
    287 elif area_subset != "none":
    288     shape_indicies = list(
--> 289         {
    290             key: selections._geography_choose[area_subset][key]
    291             for key in cached_area
    292         }.values()
    293     )
    294     if area_subset == "states":
    295         shape = set_subarea(selections._geographies._us_states, shape_indicies)

File ~/climakitae/climakitae/core/data_load.py:290, in <dictcomp>(.0)
    286     ds_region = [geom]
    287 elif area_subset != "none":
    288     shape_indicies = list(
    289         {
--> 290             key: selections._geography_choose[area_subset][key]
    291             for key in cached_area
    292         }.values()
    293     )
    294     if area_subset == "states":
    295         shape = set_subarea(selections._geographies._us_states, shape_indicies)

KeyError: 'L' 
```

Another suggestion is that we add some sort of helpful tip either to the notebook markdown (easier option, but more likely to be overlooked) or the panel itself that indicates that you can select multiple area options when holding down the command key but not the shift key. We've produced similar "red text warnings" in other panels that I think could be reproduced here too; however if others think this is intuitive and doesn't need to be stated, this can be ignored

Nice work!

File: climakitae/core/data_load.py
Comment (vicford) [reply climakitae/core/data_load.py:295]: Indices has a typo in it throughout (there's an extra "i") - minor though but would need to addressed throughout!

Comment (claalmve) [reply climakitae/core/data_load.py:295]: 🤦‍♂️ ... I am a fool... will fix.

File: climakitae/core/data_interface.py
Comment (thenaomig) [reply climakitae/core/data_interface.py:267]: cool

**Review by thenaomig — CHANGES_REQUESTED** [review]
This is great. Can approve after you address the compatibility in the other panels that inherit from data_interface. We should wait until after the refactor is merged and then redirect this as a PR to main next week.

File: climakitae/core/data_load.py
Comment (claalmve) [reply climakitae/core/data_load.py:277]: Yeah, I can change the import to `from geopandas import GeoDataFrame` instead, so it only imports the `GeoDataFrame` object instead of the whole library.

**Comment by thenaomig** [comment]
Looks all good to me :)

**Review by thenaomig — APPROVED** [review]

**Review by vicford — CHANGES_REQUESTED** [review]
Warming levels works great, as does AMY. 

Before this gets merged, can we modify the title of the AMY heatmap so that the area selection is not in a list of strings? This looks especially funny when multiple selections are chosen, and depending on the length of those strings, it does get cut off in the title (example below)
bokeh_plot

**Comment by claalmve** [comment]
> Warming levels works great, as does AMY.
> 
> Before this gets merged, can we modify the title of the AMY heatmap so that the area selection is not in a list of strings? This looks especially funny when multiple selections are chosen, and depending on the length of those strings, it does get cut off in the title (example below) bokeh_plot

Yes! I can list them out in written form, and have them indent so they don't run off of the panel screen. But I should have them all listed out, right?

**Review by elehmer — APPROVED** [review]
This is looking VERY good! Thank you!

**Pull request merged**
